ginivision/src/main/java/net/gini/android/vision/document/GiniVisionDocument.java

```diff
@@ -0,0 +1,175 @@
+package net.gini.android.vision.document;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Parcel;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import net.gini.android.vision.Document;
+import net.gini.android.vision.internal.camera.photo.ImageCache;
+import net.gini.android.vision.internal.util.UriReaderAsyncTask;
+import net.gini.android.vision.internal.util.IntentHelper;
+
+import java.util.Arrays;
+
+public class GiniVisionDocument implements Document {
+
+    private final Type mType;
+    private byte[] mData;
+    private final boolean mIsReviewable;
+    private final boolean mIsImported;
+    private final Intent mIntent;
+
+    GiniVisionDocument(@NonNull final Type type,
+            @Nullable final byte[] data,
+            @Nullable final Intent intent,
+            final boolean isReviewable,
+            final boolean isImported) {
+        mType = type;
+        mData = data;
+        mIntent = intent;
```

Comment:
I was also thinking about this, but then I thought maybe our clients want to do their own checks on the Intent for security reasons since it can come from any app. In the Review Screen or Analysis Screen they can close GVL in case they detect something suspicious.